File: VB/DevExpress.XtraReports.CustomControls/XRDataContainerStyles.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing.Design
Imports System.Linq
Imports System.Text
Imports DevExpress.XtraReports.UI
Imports DevExpress.XtraReports.Design

Namespace DevExpress.XtraReports.CustomControls
	<TypeConverter(GetType(XRControlStylesConverter)), Editor(GetType(XRControlStylesEditor), GetType(UITypeEditor))>
	Public Class XRDataContainerStyles
		Inherits DevExpress.XtraReports.UI.XRControl.XRControlStyles

		Public Sub New(ByVal owner As XRDataContainerControl)
			MyBase.New(owner)
		End Sub

		<Browsable(False), EditorBrowsable(EditorBrowsableState.Never)>
		Public Overrides Property Style() As XRControlStyle
		<Browsable(False), EditorBrowsable(EditorBrowsableState.Never)>
		Public Overrides Property EvenStyle() As XRControlStyle
		<Browsable(False), EditorBrowsable(EditorBrowsableState.Never)>
		Public Overrides Property OddStyle() As XRControlStyle

		<Editor(GetType(XRControlStyleEditor), GetType(UITypeEditor)), DefaultValue(DirectCast(Nothing, String)), DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden), RefreshProperties(RefreshProperties.All), TypeConverter(GetType(XRControlStyleConverter)), DisplayName("HeaderStyle")>
		Public Overridable Property HeaderStyle() As XRControlStyle
			Get
				Return CType(Me.control, XRDataContainerControl).HeaderStyleCore
			End Get
			Set(ByVal value As XRControlStyle)
				CType(Me.control, XRDataContainerControl).HeaderStyleCore = value
			End Set
		End Property

		<Editor(GetType(XRControlStyleEditor), GetType(UITypeEditor)), DefaultValue(DirectCast(Nothing, String)), DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden), RefreshProperties(RefreshProperties.All), TypeConverter(GetType(XRControlStyleConverter)), DisplayName("EvenCellStyle")>
		Public Overridable Property EvenCellStyle() As XRControlStyle
			Get
				Return CType(Me.control, XRDataContainerControl).EvenCellStyleCore
			End Get
			Set(ByVal value As XRControlStyle)
				CType(Me.control, XRDataContainerControl).EvenCellStyleCore = value
			End Set
		End Property

		<Editor(GetType(XRControlStyleEditor), GetType(UITypeEditor)), DefaultValue(DirectCast(Nothing, String)), DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden), RefreshProperties(RefreshProperties.All), TypeConverter(GetType(XRControlStyleConverter)), DisplayName("OddCellStyle")>
		Public Overridable Property OddCellStyle() As XRControlStyle
			Get
				Return CType(Me.control, XRDataContainerControl).OddCellStyleCore
			End Get
			Set(ByVal value As XRControlStyle)
				CType(Me.control, XRDataContainerControl).OddCellStyleCore = value
			End Set
		End Property

		<Editor(GetType(XRControlStyleEditor), GetType(UITypeEditor)), DefaultValue(DirectCast(Nothing, String)), DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden), RefreshProperties(RefreshProperties.All), TypeConverter(GetType(XRControlStyleConverter)), DisplayName("CellStyle")>
		Public Overridable Property CellStyle() As XRControlStyle
			Get
				Return CType(Me.control, XRDataContainerControl).CellStyleCore
			End Get
			Set(ByVal value As XRControlStyle)
				CType(Me.control, XRDataContainerControl).CellStyleCore = value
			End Set
		End Property
	End Class
End Namespace

File: CS/DevExpress.XtraReports.CustomControls/XRDataContainerStyles.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing.Design;
using System.Linq;
using System.Text;
using DevExpress.XtraReports.UI;
using DevExpress.XtraReports.Design;

namespace DevExpress.XtraReports.CustomControls
{
    [
    TypeConverter(typeof(XRControlStylesConverter)), 
    Editor(typeof(XRControlStylesEditor), typeof(UITypeEditor))
    ]
    public class XRDataContainerStyles : DevExpress.XtraReports.UI.XRControl.XRControlStyles
    {        
        public XRDataContainerStyles(XRDataContainerControl owner) : base(owner)
        {
        }

        [Browsable(false), EditorBrowsable(EditorBrowsableState.Never)]
        public override XRControlStyle Style { get; set; }
        [Browsable(false), EditorBrowsable(EditorBrowsableState.Never)]
        public override XRControlStyle EvenStyle { get; set; }
        [Browsable(false), EditorBrowsable(EditorBrowsableState.Never)]
        public override XRControlStyle OddStyle { get; set; }

        [
        Editor(typeof(XRControlStyleEditor), typeof(UITypeEditor)),
        DefaultValue((string)null), DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden),
        RefreshProperties(RefreshProperties.All), TypeConverter(typeof(XRControlStyleConverter)),
        DisplayName("HeaderStyle")
        ]
        public virtual XRControlStyle HeaderStyle
        {
            get
            {
                return ((XRDataContainerControl)this.control).HeaderStyleCore;
            }
            set
            {
                ((XRDataContainerControl)this.control).HeaderStyleCore = value;
            }
        }

        [
        Editor(typeof(XRControlStyleEditor), typeof(UITypeEditor)),
        DefaultValue((string)null), DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden),
        RefreshProperties(RefreshProperties.All), TypeConverter(typeof(XRControlStyleConverter)),
        DisplayName("EvenCellStyle")
        ]
        public virtual XRControlStyle EvenCellStyle
        {
            get
            {
                return ((XRDataContainerControl)this.control).EvenCellStyleCore;
            }
            set
            {
                ((XRDataContainerControl)this.control).EvenCellStyleCore = value;
            }
        }
        
        [
        Editor(typeof(XRControlStyleEditor), typeof(UITypeEditor)),
        DefaultValue((string)null), DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden),
        RefreshProperties(RefreshProperties.All), TypeConverter(typeof(XRControlStyleConverter)),
        DisplayName("OddCellStyle")
        ]
        public virtual XRControlStyle OddCellStyle
        {
            get
            {
                return ((XRDataContainerControl)this.control).OddCellStyleCore;
            }
            set
            {
                ((XRDataContainerControl)this.control).OddCellStyleCore = value;
            }
        }
        
        [
        Editor(typeof(XRControlStyleEditor), typeof(UITypeEditor)),
        DefaultValue((string)null), DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden),
        RefreshProperties(RefreshProperties.All), TypeConverter(typeof(XRControlStyleConverter)),
        DisplayName("CellStyle")
        ]
        public virtual XRControlStyle CellStyle
        {
            get
            {
                return ((XRDataContainerControl)this.control).CellStyleCore;
            }
            set
            {
                ((XRDataContainerControl)this.control).CellStyleCore = value;
            }
        }
    }
}
